File: Python solutions/58. Length of Last Word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        # words = s.split()
        # while words[-1] == '':
        #     words.pop()
        #     # print(words)

        # return len(words[-1])
        return len(s.split()[-1])
    # def lengthOfLastWord(self, s):
    
    #     wordlist = s.split()
    #     print(wordlist)
    #     if wordlist:
    #         return len(wordlist[-1])
    #     return 0

    

s1 = "Hello World"
ans1 = 5
assert Solution().lengthOfLastWord(s1) == ans1, "Test failed!"

s2 = "   fly me   to   the moon  "
ans2 = 4
assert Solution().lengthOfLastWord(s2) == ans2, "Test failed!"


print("\033[32mAll tests passed!\033[0m\n")

File: 58.Length_of_Last_Word.cc
#include <iostream>
#include <cassert>

class Solution {
public:
    int lengthOfLastWord(std::string s) {
        int ans = 0;
        bool counting = false;

        for (int idx = s.size() - 1; idx >= 0; --idx) {
            if (s[idx] != ' ') {
                counting = true;
                ++ans;
            } else if (counting) {
                break;
            }
        }

        return ans;
    }
};

/*

*/

int main() {
    Solution solution;
    std::string s1 = "Hello World";
    int ans1 = 5;
    assert(solution.lengthOfLastWord(s1) == ans1);

    std::string s2 = "   fly me   to   the moon  ";
    int ans2 = 4;
    assert(solution.lengthOfLastWord(s2) == ans2);

    std::cout << "\033[32mAll tests passed!\033[0m\n";
    return 0;
}